app/notifications/process_notifications.py

```diff
@@ -112,11 +113,14 @@ def persist_notification(
     elif notification_type == EMAIL_TYPE:
         notification.normalised_to = format_email_address(notification.to)
     elif notification_type == LETTER_TYPE:
-        template = dao_get_template_by_id(template_id, template_version)
-        if service.has_permission(CHOOSE_POSTAGE) and template.postage:
-            notification.postage = template.postage
+        if postage:
+            notification.postage = postage
         else:
-            notification.postage = service.postage
+            template = dao_get_template_by_id(template_id, template_version)
+            if service.has_permission(CHOOSE_POSTAGE) and template.postage:
+                notification.postage = template.postage
```

Comment:
The template.postage should be passed into this method, we already have the template before we call this method.  That will save us a db call.

Reply:
I can do it, but then I would have to add that argument in many places that call `persist_notification`, and check that all those places have template object:

![screen shot 2019-01-11 at 14 40 18](https://user-images.githubusercontent.com/20957548/51040277-f0cd7400-15ae-11e9-87dc-f2a2ed574e7f.png)

I didn't introduce this call to db for this particular story, I just changed indentation.

Reply:
Anyway, let me know.

Reply:
If we can avoid a call to the db we should.

Reply:
ok, this has been done now 👍